src/algorithms/tracking/TrackProjector.cc

```diff
@@ -148,7 +148,7 @@ namespace eicrecon {
                 const float pathLength = static_cast<float>(trackstate.pathLength());
                 const float pathLengthError = 0;
 
-                uint64_t surface = 0; // trackstate.referenceSurface().geometryId().value(); FIXME - ASAN is not happy with this
+                uint64_t surface = trackstate.referenceSurface().geometryId().value();
                 uint32_t system = 0;
```

Comment:
Per https://github.com/acts-project/acts/blob/020a076bebfea8f9c01a4fb0813f5bcdb495590c/Core/include/Acts/Geometry/GeometryIdentifier.hpp#L61-L62, we should use the `extra` bit fields in `geometryId` to store this subsystem ID... Which means we didn't really need the `system` field...

```suggestion
                uint32_t system = trackstate.referenceSurface().geometryId().extra();
```

This would allow us to stuff the info from https://github.com/eic/EICrecon/blob/main/src/global/tracking/TrackPropagation_factory.cc#L165-L166 into the geometry ID, making that a bit cleaner.

Reply:
#1057

Reply:
... storing the 8-bit DD4hep detector subsystem ID in the 8-bit `extra` field of `geometryId` is not as easy as I thought it would be. We can pass a GeometryIdentifierHook to Acts::convertDD4hepDetector which can write the `extra` field, but it doesn't seem to have access at the point when it is called to the DD4hep geometry objects. So, we only seem to be able to use the other `geometryId` fields to create the `extra` field, or we have to bundle an Acts::Surface-to-detector-subsystem-ID map in the hook.

And if we're going to have to create a map in ActsGeometryProvider, we might as well set the map after geometry creation and use it to set the `TrackPoint::system` field here based on the `volume` field in `geometryId` IFF `extra` is not set.

Reply:
Sure, I was talking about this when `system` field was added.

Reply:
Sorry, must have missed that. Figured out the GeometryIdentifierHook in #1058.

Reply:
The `GeometryIdentifierHook` I didn't know about at the time. I thought there was no facility to actually support this.

Reply:
It certainly isn't well advertised. But it seems to have been added exactly for this purpose.

Reply:
Yeah, nice find.